### ISSUE TITLE
Heh, yeah I guess tractor.run() is kinda dumb..

### ISSUE THREAD

**Issue by goodboy** [goodboy/tractor#177]
This is somewhat of a 🤦🏼 and directly related to #168.

Thanks to the `trio` crowd for pointing out that we can probably just make ourselves a:

```python

import tractor
import trio

async main():
    async with tractor.open_runtime():
        .. do tractor things since this is a task...
        
trio.run(main)
```

Some benefits to this:
- it's not scary since it's `trio.run()` 🤯 
- we can run certain parts of code with `debug_mode=True` ?
- the whole process tree can be restarted easily from scratch 
- we aren't hijacking `trio.run()` when we really have no reason to

Some worries:
- starting multiple "runtimes" (aka root actors) in a single program will cause mailbox collisions
  - maybe we could have a `.open_root_runtime()` which can only be called once per program?
  - is this solved by just avoiding channel server address collisions?
- maybe more?

Is there a better name?
- `.open_root()`
- `.start_runtime()`
- `start_root()`

Lurkerzzzz!!!

**Comment by richardsheridan** [comment]
IMHO the ideal API would be:

```python
async with tractor.open_actor_nursery() as n:
    n.start_soon(async_fn)

```

And tractor should take care of any global bookkeeping internally, if there is such a thing.  $0.02 :)

**Comment by goodboy** [comment]
@richardsheridan only problem is there might only ever be *one* "actor_nursery* open per program.

Also note, in current parlance, an `ActorNursery` is a nursery [which spawns actors](https://github.com/goodboy/tractor/blob/master/tractor/_trionics.py#L25) - aka known as new processes running `trio.run()`. So I don't think calling the top level entry point for starting all the machinery/runtime that makes actor spawning possible a "nursery" necessarily makes sense.

maybe `.open_root_actor()`?

**Comment by goodboy** [comment]
Hmm thinking more on this I wonder if we can just do implicit runtime startup (only in the root process) on the first entry to `tractor.open_nursery()`?

I'm thinking there should still be an explicit `.start_root_actor_runtime()` or whatever but that can be implicitly called if we find `tractor.is_root_process() == True` when entering the `.open_nursery()` block.

**Comment by goodboy** [comment]
The removal of `tractor.run()` is going to require a big edit of the test set and examples.
- [ ]  [`tractor_test`](https://github.com/goodboy/tractor/blob/master/tractor/testing/_tractor_test.py#L57) is going to need to be redone
- [ ]  the boatload of [tests still using it](https://github.com/search?q=tractor.run+repo%3Agoodboy%2Ftractor+path%3Atests&type=Code)
- [ ] all the [docs examples](https://github.com/search?q=tractor.run+repo%3Agoodboy%2Ftractor+path%3Aexamples&type=Code&ref=advsearch&l=&l=)

**Comment by goodboy** [comment]
Resolved with #197.
Full `.run()` removal will come in a upcoming release.
